Task/table states

MichaelNash1990 opened the pull request:
- Add table row and table row data to a database.
- Record permissions on their own table instead of on the actions table.
- Record abi data on abis table.
- Perform some general refactoring of existing code.
- Added configuration for API: IP-address:PORT
- Increased performance by an average 200-300%.
- Ridded of all psql-external commands.
- Cleaned the code.

MichaelNash1990 committed:
EX-1164 Add table row and table row data code

MichaelNash1990 committed:
EX-1164 Fix compiler error

MichaelNash1990 committed:
EX-1164 Create troubleshooting print statement

MichaelNash1990 committed:
EX-1164 Write table row data to database

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to resolve compilation problems

MichaelNash1990 committed:
EX-1164 Attempt to record table row data

MichaelNash1990 committed:
EX-1164 Fix problem with table row number global variable

MichaelNash1990 committed:
EX-1164 Fix problem with table row number global variable

MichaelNash1990 committed:
EX-1164 Fix problem with table row number global variable

MichaelNash1990 committed:
EX-1164 Fix problem with table row number global variable

MichaelNash1990 committed:
EX-1164 Create new vector to avoid altering the values for the table_row_data table

MichaelNash1990 committed:
EX-1164 Fix compiler error

MichaelNash1990 committed:
EX-1164 Convert delta data to hex and print it

MichaelNash1990 committed:
EX-1164 Attmept to resolve compiler errors

MichaelNash1990 committed:
EX-1164 Attmept to resolve compiler errors

MichaelNash1990 committed:
EX-1164 Attmept to resolve compiler errors

MichaelNash1990 committed:
Fix indentation

MichaelNash1990 committed:
Add the mandel abieos library

MichaelNash1990 committed:
Save progress on creating an abieos context

MichaelNash1990 committed:
Add abi context, add abi to context whenever setabi is called

MichaelNash1990 committed:
Use abieos instead of mandel-abieos

MichaelNash1990 committed:
Don't process setcode action data

MichaelNash1990 committed:
Write compiling code for recording table states using mandel-abieos

MichaelNash1990 committed:
Use delta name instead of table name for if statement

MichaelNash1990 committed:
Create debugging code

MichaelNash1990 committed:
Continue debugging

MichaelNash1990 committed:
Remove commit line

MichaelNash1990 committed:
Resolve issues with connecting to database, eosio::context not being populated

MichaelNash1990 committed:
Identify that the contexts being created are different

MichaelNash1990 committed:
Populate database with table row data

MichaelNash1990 committed:
Rework some table data

MichaelNash1990 committed:
Add try catch statements

MichaelNash1990 committed:
Add extra TODO statement

MichaelNash1990 committed:
Add new permissions table, add elog and throw statements inside catch

MichaelNash1990 committed:
Fix table column types

MichaelNash1990 committed:
Fix indentation

MichaelNash1990 committed:
Add extra TODO

MichaelNash1990 committed:
Change variable types from VARCHAR to CHAR

MichaelNash1990 committed:
Search differently for existing abi

MichaelNash1990 committed:
fix conflicts

MichaelNash1990 committed:
Add abi table

MichaelNash1990 committed:
Resolve issue with abis not being loaded into context correctly upon program restart

MichaelNash1990 committed:
Work on adding foreign keys to created tables

MichaelNash1990 committed:
Add index for tables

MichaelNash1990 committed:
Delete dead code, fix formatting

MichaelNash1990 committed:
Delete unneeded print statements

shannendorothee commented on src/fill_pg_plugin.cpp:
```diff
@@ -266,40 +321,65 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
+      t.exec("create index permissions_action_number on " + converter.schema_name + ".permissions" + R"((action_number))");
+      t.exec("create index permissions_actor on " + converter.schema_name + ".permissions" + R"((actor))");
+      t.exec("create index abis_action_number on " + converter.schema_name + ".abis" + R"((action_number))");
+      t.exec("create index abis_account on " + converter.schema_name + ".abis" + R"((account))");
```
maybe lines 354-375 can be shortened through looping a list? seems repetitive

MichaelNash1990 replied:
@shannendorothee This is definitely a good suggestion - I could for instance create a map, then iterate through the map with a loop function. But I think in this case the map would end up using up just as many lines of code as it is right now, in order to maintain readability. So for now I think I'll leave it as it is

AleksandrAksenov replied:
Updated.

shannendorothee commented on src/fill_pg_plugin.cpp:
```diff
@@ -602,28 +679,50 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
+        }
+        else if (name == "actions")
+        {
+          columns = {"action_number", "transaction_number", "action_ordinal", "creator_action_ordinal", "receiver", "action_account", "action_name", "context_free", "console"};
```
maybe initialise these column names and also on line 691 in a separate file, use it as a schema

MichaelNash1990 replied:
@shannendorothee This is a good suggestion. As it stands, there are tons of refactors we could implement like this, mostly due to the fact that the project this was forked from is open source.

It would be great to improve the whole codebase as it stands, but we don't plan to alter this project once we're done with it for quite some time, so I'm not sure it's worth taking the time to extract code like this to their own separate files. Especially since we need to focus our efforts on the advanced exchange right now.

AleksandrAksenov replied:
updated

AleksandrAksenov committed:
Prettified the code
Removed duplicated execution of every command
Refactored tables indexes creation
Refactored columns condition

AleksandrAksenov committed:
Prettified the code

AleksandrAksenov committed:
Added fill-api argument for cleos command. Format: IP-ADDRESS:PORT (ex.
127.0.0.1:8888)
Added SQL connection for select queries.
Ridded off all psql commands execution.
Removed duplicated command execution.
Refactored create index function.
Refactored write_stream_custom_columns.

AleksandrAksenov committed:
Fixed erased order

AleksandrAksenov committed:
Cleaned CMakeLists.txt

AleksandrAksenov committed:
Fixed hex_data type

AleksandrAksenov committed:
Refactored values vectors initialization

AleksandrAksenov committed:
Refactored host/port assignment

MichaelNash1990 commented on src/fill_pg_plugin.cpp:
```diff
@@ -808,11 +763,11 @@ struct fpg_session : connection_callbacks, std::enable_shared_from_this<fpg_sess
+	    values.erase(values.begin() + 4);
+	    values.erase(values.begin() + 8);
+	    values.erase(values.begin() + 9);
+	    values.erase(values.begin() + 10);
```
@AleksandrAksenov I didn't finish reviewing your changes yet, but are you sure this works? If you remove the 4th element first, the 10th element will be different by the time you wish to delete it, right?

MichaelNash1990 replied:
@AleksandrAksenov Never mind, I saw you changed it back.

AleksandrAksenov replied:
You are right - I accidentally set the wrong order and later changed it to the correct one.
